src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -367,6 +376,59 @@ public function getDefinedOptions()
     }
 
     /**
+     * Freeze the option with the given name.
+     *
+     * @param string|string[] $optionNames One or more option names
+     *
+     * @return OptionsResolver This instance
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function setFrozen($optionNames)
+    {
+        if ($this->locked) {
+            throw new AccessException('Options cannot be freeze from a lazy option or normalizer.');
+        }
+
+        foreach ((array) $optionNames as $option) {
+            if (!$this->hasDefault($option)) {
+                throw new MissingDefaultValueException(sprintf(
+                    'The option "%s" has not a default value. You can not freeze it.',
```

Comment:
I suggest using shorter `has no default value`. In the next sentence, I'd also use the more usual form `cannot` instead of `can not`.